src/praisonai/praisonai/train/llm/trainer.py

```
@@ -10,18 +10,38 @@
+            'FastLanguageModel': FastLanguageModel,
+            'is_bfloat16_supported': is_bfloat16_supported,
+            'SFTTrainer': SFTTrainer,
+            'TrainingArguments': TrainingArguments,
+            'load_dataset': load_dataset,
+            'concatenate_datasets': concatenate_datasets,
+            'virtual_memory': virtual_memory,
+            'standardize_sharegpt': standardize_sharegpt,
+            'get_chat_template': get_chat_template,
+        })
+    except ImportError as e:
+        raise ImportError(f"Training dependencies not available. Install with: pip install torch transformers unsloth datasets trl psutil. Error: {e}")
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)